diff --git a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
index 2daec5d..124078e 100644
--- a/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
+++ b/node_modules/expo/ios/AppDelegates/EXReactRootViewFactory.h
@@ -8,6 +8,7 @@
 NS_ASSUME_NONNULL_BEGIN
 
 @class EXReactDelegate;
+@class RCTDevMenuConfiguration;
 
 NS_SWIFT_NAME(ExpoReactRootViewFactory)
 @interface EXReactRootViewFactory : RCTRootViewFactory
